dfget/core/core.go

```diff
@@ -299,7 +299,7 @@ func reportMetrics(cfg *config.Config, supernodeAPI api.SupernodeAPI, downloadTi
 		if err != nil {
 			logrus.Errorf("failed to report metrics to supernode %s: %v", node, err)
 		}
-		if resp.IsSuccess() {
+		if resp != nil && resp.IsSuccess() {
```

Comment:
What about moving this nil check to method resp.IsSuccess?  If resp is nil, then return false.  IMO it is more clear.